### PR TITLE
added support for FreeBSD

### PR DESCRIPTION
**Pull request by blingu**
Tested on 12.1-RELEASE

### CODE REVIEW

File: src/core/CdStreamPosix.cpp [hunk 1]
```diff
@@ -150,9 +150,11 @@ CdStreamInit(int32 numChannels)
         ASSERT(0);
         return;
     }
-
+#ifdef __linux__
 	_gdwCdStreamFlags = O_RDONLY | O_NOATIME;
-
+#elif __FreeBSD__
```
Comment (erorcun) [inline src/core/CdStreamPosix.cpp:155]: you can do this just #else

File: src/skel/glfw/glfw.cpp [hunk 2]
```diff
@@ -209,7 +209,11 @@ double
 psTimer(void)
 {
 	struct timespec start; 
+#ifdef __linux__
 	clock_gettime(CLOCK_MONOTONIC_RAW, &start);
+#elif __FreeBSD__
```
Comment (erorcun) [inline src/skel/glfw/glfw.cpp:214]: same here

File: premake5.lua [hunk 3]
```diff
@@ -1,308 +1,328 @@
-newoption {
```
Comment (erorcun) [inline premake5.lua:1]: can you fix the formatting pls? :/

File: src/core/CdStreamPosix.cpp (hunk 1 above)
Comment (blingu) [reply src/core/CdStreamPosix.cpp]: fixed

File: src/skel/glfw/glfw.cpp (hunk 2 above)
Comment (blingu) [reply src/skel/glfw/glfw.cpp]: fixed

File: premake5.lua (hunk 3 above)
Comment (blingu) [reply premake5.lua]: fixed

**Pull request merged**